src/Workspaces/Remote/ServiceHub/Services/LegacySolutionEvents/RemoteLegacySolutionEventsAggregationService.cs

```diff
@@ -54,9 +60,46 @@ public ValueTask OnTextDocumentClosedAsync(Checksum solutionChecksum, DocumentId
                 async (oldSolution, newSolution) =>
                 {
                     var aggregationService = oldSolution.Services.GetRequiredService<ILegacySolutionEventsAggregationService>();
-                    await aggregationService.OnWorkspaceChangedEventAsync(
-                        new WorkspaceChangeEventArgs(kind, oldSolution, newSolution, projectId, documentId), cancellationToken).ConfigureAwait(false);
+                    await aggregationService.OnWorkspaceChangedAsync(
+                        GetDescriptor(), new WorkspaceChangeEventArgs(kind, oldSolution, newSolution, projectId, documentId), cancellationToken).ConfigureAwait(false);
                 }, cancellationToken);
         }
+
+        /// <summary>
+        /// NOTE: When we remove the RemoteWorkspace from OOP this will be impacted.  The approach we should take then
+        /// is that this descriptor works by trying to return the latest primary-solution snapshot that VS has sync'ed
+        /// to OOP.  This is invariably an unpleasant concept in OOP (which we want to be entirely checksum/snapshot
+        /// based).  However, this unpleasantness is necessary as this code exists *solely* to allow UnitTesting to
+        /// continue running with their own form of SolutionCrawler, and SolutionCrawler's design itself is deeply
+        /// connected to the concept of the latest "CurrentSolution".
+        /// 
+        /// Generally speaking, this should still continue to work though as we will still very likely have it as part
+        /// of our design that OOP holds onto the latest-primary-solution-snapshot as that's still very helpful for perf
+        /// reasons, even if the individual OOP requests come in are referencing a specific checksum/snapshot pair.  The
+        /// caching of the latest-primary-solution-snapshot helps keep appropriate assets pinned in OOP and makes
+        /// producing the requested snapshot cheaper is it can be a fork of that version we are holding onto.
+        /// 
+        /// In other words, we are not keeping around the latest-primary-solution-snapshot to power UnitTesting
+        /// solution-crawling.  We are keeping it around for perf around standard OOP snapshot requests, and this
+        /// component can appropriately leverage that.
+        /// </summary>
```

Comment:
@tmat The above contains my thinking on why this direction is ok.  This is jsut a super unfortunate aspect of the design of solution-crawler.  However, it does feel like it's something we can support even as we move off of RemoteWorkspace in OOP.

Reply:
The above does require is to effectively 'expose' that we are caching teh 'latest primary solution snapshot' in OOP (outside of hte syncing system).  It's not ideal, but it's also hopefully very temporary esp. as we continue to push on unittesting to get off of solution-crawler (even their fork of it).

Reply:
Couldn't solution crawler figure what the current solution is from the solution events? Every time in-proc workspace current solution is updated we send an event to OOP which has the snapshot, right?

Reply:
I.e. the solution crawler would track what the current snapshot is instead of remote workspace.

Reply:
> Couldn't solution crawler figure what the current solution is from the solution events? Every time in-proc workspace current solution is updated we send an event to OOP which has the snapshot, right?

Maybe.................................

It does seem reasonable.   And i think i *will* change to that in the next PR.   Though it def makes me nervous that this will somehow be different due to some crazy aspect of SolutionCrawler taht we don't understand :D  

I'm very torn on this work between trying to keep things as absolutely close to how SolutionCrawler works today, while also wanting to make judicious changes to keep things as unhacky as possible.  This feels like it's on the fence.  But worthwhile enough to do.

Reply:
> I.e. the solution crawler would track what the current snapshot is instead of remote workspace.

Yup.  Understood.  And i think taht is reasonable given that only the 'host' issues the stream of workspace events, and that happens in a serialized queue.  So we always just reocrd the latest and have that be conceptually the same as the one the RemoteWorkspace is holding onto.